### PR TITLE
Prow integration test: add hook test

### PR DESCRIPTION
**Pull request by chaodaiG**
Enable prow integration test covering hook. This PR
- Add hook deployment in prow integration test
- Injecting webhook events to hook in kind cluster
- Fakeghserver supports labels operations.

Fun fact: the plugin `label` was picked expecting it doesn't involve too much fakeghserver changes, unfortunately turned out to be not quite the case.

### CODE REVIEW

**Comment by chaodaiG** [comment]
/area prow
/assign @cjwagner @alvaroaleman 

/cc @spiffxp 
Since this uses webhook events from spiffxp :)

**Comment by chaodaiG** [comment]
@cjwagner @alvaroaleman please take a look when you got a chance

**Comment by chaodaiG** [comment]
> @cjwagner @alvaroaleman please take a look when you got a chance

ping

File: prow/test/integration/fakeghserver/fakeghserver.go [hunk 1]
```diff
@@ -226,3 +233,54 @@ func issueCommentHandler(ghc *fakegithub.FakeClient) func(*http.Request) (string
 		return "", http.StatusInternalServerError, fmt.Errorf("{\"error\": \"API not supported\"}, %s, %s", r.URL.Path, r.Method)
 	}
 }
+
+//  GetRepoLabels(org, repo string) ([]Label, error) # fmt.Sprintf("/repos/%s/%s/labels", org, repo),
+//  GetIssueLabels(org, repo string, number int) ([]Label, error) # fmt.Sprintf("/repos/%s/%s/issues/%d/labels", org, repo, number)
+//  AddLabel(org, repo string, number int, label string) # fmt.Sprintf("/repos/%s/%s/issues/%d/labels", org, repo, number)
+//  AddRepoLabel(org, repo, label, description, color string) error # fmt.Sprintf("/repos/%s/%s/labels", org, repo),
+func labelHandler(ghc *fakegithub.FakeClient) func(*http.Request) (interface{}, int, error) {
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply prow/test/integration/fakeghserver/fakeghserver.go:241]: Part of it was laziness :). Overall it does save some code though, for example `org, repo := vars["org"], vars["repo"]` and `id, err := strconv.Atoi(issueID)
			if err != nil {
				return "", http.StatusUnprocessableEntity, err
			}`

File: prow/test/integration/test/examples/test_comment.json [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    "downloads_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/downloads",
+    "issues_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/issues{/number}",
+    "pulls_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/pulls{/number}",
+    "milestones_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/milestones{/number}",
+    "notifications_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/notifications{?since,all,participating}",
+    "labels_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/labels{/name}",
+    "releases_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/releases{/id}",
+    "deployments_url": "https://api.github.com/repos/fake-org-hook/fake-repo-hook/deployments",
+    "created_at": "2016-04-28T21:05:35Z",
+    "updated_at": "2016-11-23T18:03:45Z",
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply prow/test/integration/test/examples/test_comment.json]: Updated with a much up-to-date webhook payload

File: prow/test/integration/prow/cluster/hook_deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      containers:
+      - name: hook
+        image: localhost:5000/hook
+        imagePullPolicy: Always
+        args:
+        - --dry-run=false
+        - --github-endpoint=http://fakeghserver
+        - --github-token-path=/etc/github/oauth
+        - --config-path=/etc/config/config.yaml
+        - --plugin-config=/etc/plugins/plugins.yaml
```
Comment (cjwagner) [inline prow/test/integration/prow/cluster/hook_deployment.yaml:48]: Still missing the HMAC flag.

**Review by cjwagner — COMMENTED** [review]
One last nit otherwise this LGTM.

File: prow/test/integration/prow/cluster/hook_deployment.yaml (hunk 3 above)
Comment (chaodaiG) [reply prow/test/integration/prow/cluster/hook_deployment.yaml:48]: done

**Review by cjwagner — APPROVED** [review]

**Pull request merged**